lib/src/map/options/interaction_options.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:flutter/services.dart';
+import 'package:flutter_map/src/map/options/enabled_gestures.dart';
+import 'package:meta/meta.dart';
+
+/// Set interation options for input gestures.
+/// Most commonly used is [InteractionOptions.enabledGestures].
+@immutable
+final class InteractionOptions {
+  /// Enable or disable specific gestures. By default all gestures are enabled.
+  /// If you want to disable all gestures or almost all gestures, use the
+  /// [EnabledGestures.none] constructor.
+  /// In case you want to disable only few gestures, use [EnabledGestures.all]
+  /// and you can use the [EnabledGestures.noRotation] for an easy way to
+  /// disable all rotation gestures.
+  ///
+  /// In addition you can specify your gestures via bitfield operations using
+  /// the [EnabledGestures.bitfield] constructor together with the static
+  /// fields in [InteractiveFlag].
+  /// For more information see the documentation on [InteractiveFlag].
+  final EnabledGestures enabledGestures;
+
+  /// Map starts to rotate when [twoFingerRotateThreshold] has been achieved
+  /// or another multi finger gesture wins.
+  /// Default is 0.1
+  final double twoFingerRotateThreshold;
+
+  /// Map starts to zoom when [twoFingerZoomThreshold] has been achieved or
+  /// another multi finger gesture wins.
+  /// Default is 0.1
+  final double twoFingerZoomThreshold;
+
+  /// Map starts to move when [twoFingerMoveThreshold] has been achieved or
+  /// another multi finger gesture wins. This doesn't take any effect on drag
+  /// gestures by a single pointer like a single finger.
+  ///
+  /// This option gets superseded by [twoFingerZoomThreshold] if
+  /// [EnabledGestures.twoFingerMove] and [EnabledGestures.twoFingerZoom] are
+  /// both active and the [twoFingerZoomThreshold] is reached.
+  ///
+  /// Default is 3.0 (disabled)
+  final double twoFingerMoveThreshold;
+
+  /// The velocity how fast the map should zoom when using the scroll wheel
+  /// of the mouse.
+  /// Defaults to 0.01.
+  final double scrollWheelVelocity;
+
+  /// Override this option if you want to use custom keys for the CTRL+drag
+  /// rotate gesture.
+  /// By default the left and right control key are both used.
+  final List<LogicalKeyboardKey> ctrlRotateKeys;
```

Comment:
This was one of the things I was looking to avoid in my early unfinished rewrites from a while back. Ideally, there should be no configuration of a gesture seperate to where that gesture is enabled or disabled.

Maybe that means every gesture flag/option needs to not be a boolean anymore, and needs to be an entire object `GestureOptions`, with inheritable constructors `enabled()` and `disabled()`, and each option can have it's own arguments. In combination with the recommendation above to establish better grouping of options, this would allow for 'labels'/'tags' of categories, or something similar. I'm aware of this potential impact on performance here though of having loads of new objects, so am open to benchmarks and/or alternative suggestions.

Reply:
After thinking about it my mind is still split. 

#### Pro gesture options as separate classes
- All gesture options of a gesture are separated from other gesture options
- Option to enable or disable complete gesture groups

#### Contra gesture options as separate classes
- Separation between enabled options and option overrides
- Huge class overhead and more complex map options (13+ classes, 9 additional instances)
- More complex implementation of the map options for users 
    - Either users need to save the gesture options as field variables
    - or something like `gestureEnabeld ? SomeGesture.enabled(additionalOption: 123) : SomeGesture.disabled()`

Reply:
> gestureEnabeld ? SomeGesture.enabled(additionalOption: 123) : SomeGesture.disabled()

To avoid this we can create an unnamed constructor with all additional options, and an `enabled` flag.

> In combination with the recommendation above to establish better grouping of options, this would allow for 'labels'/'tags' of categories, or something similar.

I've realised this statement, whilst true, probably won't help on the users side. Your suggestion in https://github.com/fleaflet/flutter_map/pull/1733#discussion_r1429233040 is still probably the best.

Reply:
I did some prototyping. This is what I came up with:
#### each gesture one by one:
```dart
// option 1
const opt1 = InteractionOptions(
  enabledGestures: EnabledGestures.all(
    flingAnimation: flingEnabed,
  ),
  ctrlRotateKeys: [
    ...InteractionOptions.defaultCtrlRotateKey,
    LogicalKeyboardKey.keyA,
  ],
  scrollWheelVelocity: 1,
  twoFingerMoveThreshold: 2,
  twoFingerRotateThreshold: 3,
  twoFingerZoomThreshold: 4,
);

// option 2
final opt2 = const InteractionOptions.all(
  ctrlRotate: CtrlTriggerRotateGesture.enabled(
    triggerKeys: [
      ...InteractionOptions.defaultCtrlRotateKey,
      LogicalKeyboardKey.keyA,
    ],
  ),
  scrollWheelZoom: ScrollWheelZoomGesture.enabled(velocity: 1),
  twoFingerMove: TwoFingerMoveGesture.enabled(threshold: 2),
  twoFingerRotate: TwoFingerRotateGesture.enabled(threshold: 3),
  twoFingerZoomThreshold: TwoFingerZoomThreshold.enabled(threshold: 4),
  fling: FlingAnimationGesture(enabled: flingEnabed),
  // all other gestures enabled by the .all() constructor
);
```

#### Using grouped gestures
```dart
// option 1
const opt1 = InteractionOptions(
  enabledGestures:
      EnabledGestures.byGroup(move: true, zoom: false, rotate: false),
  scrollWheelVelocity: 1,
  twoFingerMoveThreshold: 2,
  twoFingerZoomThreshold: 4,
);

// option 2
final opt2 = const InteractionOptions.byGroups(
  zoom: ZoomGestures(
    scrollWheelVelocity: 1,
    twoFingerZoomThreshold: 4,
  ),
  move: MoveGestures(
    twoFingerMoveThreshold: 2,
  ),
  rotate: RotateGestures.disabled(),
);

```

Is this option 2 the the design idea you had?

Reply:
As further discussed internally we are keeping it like this for this pull request with the option to make changes to it later on if a better design comes up than the current.